sdk/compiler/damlc/lib/DA/Cli/Damlc/Test.hs

```diff
@@ -76,10 +77,24 @@ data CoveragePaths = CoveragePaths
 newtype LoadCoverageOnly = LoadCoverageOnly {getLoadCoverageOnly :: Bool}
 
 -- | Test a Daml file.
-execTest :: SdkVersioned => [NormalizedFilePath] -> RunAllTests -> ShowCoverage -> UseColor -> Maybe FilePath -> Options -> TableOutputPath -> TransactionsOutputPath -> CoveragePaths -> [CoverageFilter] -> IO ()
-execTest inFiles runAllTests coverage color mbJUnitOutput opts tableOutputPath transactionsOutputPath resultsIO coverageFilters = do
+execTest
+    :: SdkVersioned
+    => [NormalizedFilePath]
+    -> RunAllTests
+    -> ShowCoverage
+    -> UseColor
+    -> Maybe FilePath
+    -> Maybe PackageConfigFields
+    -> Options
+    -> TableOutputPath
+    -> TransactionsOutputPath
+    -> CoveragePaths
+    -> [CoverageFilter]
+    -> IO ()
+execTest inFiles runAllTests coverage color mbJUnitOutput mPkgConfig opts tableOutputPath transactionsOutputPath resultsIO coverageFilters = do
     loggerH <- getLogger opts "test"
-    withDamlIdeState opts loggerH diagnosticsLogger $ \h -> do
+    let optsWithPkg = maybe opts (\PackageConfigFields{..} -> opts { optMbPackageName = Just pName, optMbPackageVersion = pVersion }) mPkgConfig
```

Comment:
A case statement here would be nice, similarly to what you did in response to Sam's comment about MaybeT in https://github.com/digital-asset/daml/pull/21376/files#r2156376407